### PR TITLE
Update Examples URL in documentation of master version

### LINKED ISSUE #1322: URL of examples zip file does not get updated when build script is run
### Description of issue

While doing a release, `tools/build_release.py --build` is run. This is supposed to update the URL of examples zip file with the version to be released. But that is not happening because the VERSION file is being updated after the URL is generated.

### LINKED ISSUE #1326: Master branch of documentation does not point to master version of examples
### Description of issue

On [master branch](https://nimi-python.readthedocs.io/en/master/niscope/examples.html) of ReadTheDocs, the examples URL points to released examples zip file. We should improve the documentation by adding info about where to find the master version of examples.

### PR DESCRIPTION
**Pull request by sbethur**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

~- [ ] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~

~- [ ] I've added tests applicable for this pull request~

### What does this Pull Request accomplish?

- In the master version of documentation, update the text to provide links to two versions of examples:
   1. master version
    2. latest release version
- Fix the dependency of examples.rst in `rules.mak` so that the examples URL gets updated when making a release
- Update the build script to instruct developer to update `LASTEST_RELEASE` file before `--build` step

### List issues fixed by this Pull Request below, if any.

TODO: List of issues.

* Fix #1322
* Fix #1326

### What testing has been done?

Visual inspection of generated `examples.html`

### CODE REVIEW

File: build/templates/examples.rst.mako [hunk 1]
```diff
@@ -21,15 +24,15 @@
...
     v = Version(global_version)
     if v.dev is None and v.pre is None:
-        url = 'https://github.com/ni/nimi-python/releases/download/{0}/{1}_examples.zip'.format(global_version, module_name)
+        examples_link_text = '`You can download all {0} examples here <{1}>`_'.format(module_name, released_url)
     else:
-        with open('./LATEST_RELEASE') as vf:
-            latest_release_version = vf.read().strip()
-        url = 'https://github.com/ni/nimi-python/releases/download/{0}/{1}_examples.zip'.format(latest_release_version, module_name)
+        master_url = 'https://github.com/ni/nimi-python/tree/master/src/{}/examples'.format(module_name)
+        examples_link_text = '`You can download all {0} examples for master version here <{1}>`_'.format(module_name, master_url)
```
Comment (marcoskirsch) [inline build/templates/examples.rst.mako]: We had agreed to link to the latest release. This is what 90+% of clients will have on their systems since most won't go through the trouble of cloning and building master. And those that do already have the examples locally!

Why this change?

File: docs/nidcpower/examples.rst [hunk 2]
```diff
@@ -1,7 +1,9 @@
 Examples
 ========
 
-`You can download all nidcpower examples here <https://github.com/ni/nimi-python/releases/download/1.2.0/nidcpower_examples.zip>`_
+`You can download all nidcpower examples for master version here <https://github.com/ni/nimi-python/tree/master/src/nidcpower/examples>`_
```
Comment (marcoskirsch) [inline docs/nidcpower/examples.rst]: I would remove this, it will only cause confusion. Most clients don't know nor need to know that the code is in GitHub and there is a master version. Most of the time examples won't change anywhay between master and latest release.

File: docs/nidcpower/examples.rst [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
+
+`You can download all nidcpower examples for latest version here <https://github.com/ni/nimi-python/releases/download/1.2.0/nidcpower_examples.zip>`_
```
Comment (marcoskirsch) [inline docs/nidcpower/examples.rst]: I do like saying "for the laters release".
You could even specify which version:

> You can download all nidcpower 1.2.0 examples here

File: build/templates/examples.rst.mako (hunk 1 above)
Comment (sbethur) [reply build/templates/examples.rst.mako]: Motivation for the change was to fix the broken logic in URL generation. While fixing, I thought we could be explicit and provide links to both `latest release` and `master`.

Comment (sbethur) [reply build/templates/examples.rst.mako]: I removed the link to master version of examples, to cater to 90% and avoid confusion

File: docs/nidcpower/examples.rst (hunk 2 above)
Comment (sbethur) [reply docs/nidcpower/examples.rst]: Removed

File: docs/nidcpower/examples.rst (hunk 3 above)
Comment (sbethur) [reply docs/nidcpower/examples.rst]: Currently not all the drivers have the same latest version. For such drivers, the latest version does not match the version in examples URL. Explicitly mentioning the version would cause unnecessary confusion.
